src/main/java/mate/academy/spring/controller/AuthenticationController.java

```diff
@@ -0,0 +1,37 @@
+package mate.academy.spring.controller;
+
+import mate.academy.spring.mapper.DtoResponseMapper;
+import mate.academy.spring.model.User;
+import mate.academy.spring.model.dto.request.UserRequestDto;
+import mate.academy.spring.model.dto.response.UserResponseDto;
+import mate.academy.spring.security.AuthenticationService;
+import mate.academy.spring.service.UserService;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class AuthenticationController {
+    private final AuthenticationService authenticationService;
+    private final UserService userService;
+    private final DtoResponseMapper<UserResponseDto, User> userDtoResponseMapper;
+
+    public AuthenticationController(AuthenticationService authenticationService,
+                                    UserService userService,
+                                    DtoResponseMapper<UserResponseDto, User>
+                                            userDtoResponseMapper) {
+        this.authenticationService = authenticationService;
+        this.userService = userService;
+        this.userDtoResponseMapper = userDtoResponseMapper;
+    }
+
+    @PostMapping("/register")
+    public UserResponseDto register(@RequestBody UserRequestDto userRequestDto) {
+        User user = userService.findByEmail(userRequestDto.getEmail()).orElse(null);
+        if (user == null) {
```

Comment:
you can use `isPresent()` or `isEmpty()` to check is user null or not. It's better